### PR TITLE
S2at2 code new entry form

### PR DESCRIPTION
**Pull request by MotNosniktaw**

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -18,25 +22,41 @@
...
-        <div class="item-detail country-column">Country</div>
-        <div class="item-detail value-column">Value</div>
-        <div class="item-detail date-acq-column">Date Acquired</div>
+    <div>
+        <h1>Guitar Collection</h1>
+        <p>These are the things in my collection:</p>
+    </div>
+    <div>
+        <form action="index.php" method="get">
+            <div>Brand:    <input type="text" name="brand" value="<?php echo $_GET['brand']?>"></div>
```
Comment (FelixLetheren) [inline index.php]: can use 'required' to make sure form will only send when required fields are filled

**Review by FelixLetheren — CHANGES_REQUESTED** [review]

File: functions.php [hunk 2]
```diff
@@ -78,3 +78,63 @@ function displayGuitars(array $guitarsArray):string {
     }
     return $result;
     }
```
Comment (sjr-mayden) [inline functions.php:80]: This bracket should be exdented

File: functions.php [hunk 3]
```diff
@@ -78,3 +78,63 @@ function displayGuitars(array $guitarsArray):string {
...
+
+function addNewGuitar(\PDO $database):string {
+
+    $message = '<p>I want to add a new guitar for you!</p>';
+    $brandInDB = checkBrands($database);
```
Comment (sjr-mayden) [inline functions.php]: Functions calling other functions was the problem that Cat had when Charlie reviewed her code and it required a major rewrite

File: index.php [hunk 4]
```diff
@@ -5,6 +5,18 @@
 $db = getDB();
 $allGuitars = getAllFromDatabase($db);
 
+if (isset($_GET['newGuitar'])
```
Comment (sjr-mayden) [inline index.php]: `) {` should go on a new line

File: index.php [hunk 5]
```diff
@@ -18,25 +30,48 @@
...
-        <div class="item-detail country-column">Country</div>
-        <div class="item-detail value-column">Value</div>
-        <div class="item-detail date-acq-column">Date Acquired</div>
+    <div>
+        <h1>Guitar Collection</h1>
+        <p>These are the things in my collection:</p>
+    </div>
+    <div>
+        <form action="index.php" method="get">
+            <div>Brand:    <input type="text" name="brand" value="<?php echo $_GET['brand']?>" required></div>
```
Comment (sjr-mayden) [inline index.php]: What's this extra whitespace doing here?

**Review by sjr-mayden — CHANGES_REQUESTED** [review]

File: addNewGuitar.php [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        $country = getCountryID($db);
+
+        addNewGuitar($db, $brand, $type, $country);
+        addNewImage($db);
+    } else {
+        $message .= '<p>You should add better information</p>';
+    }
+}
+?>
+<html>
```
Comment (sjr-mayden) [inline addNewGuitar.php]: Missing a doctype

File: addNewGuitar.php [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+<head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <title>Guitar Collection - Add</title>
+    <link href="https://fonts.googleapis.com/css?family=Catamaran&display=swap"
```
Comment (sjr-mayden) [inline addNewGuitar.php]: Move attributes for consistency

File: addNewGuitar.php [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+          rel="stylesheet">
+    <link rel="stylesheet" type="text/css" href="stylesheets/normalize-8-0-1.css">
+    <link rel="stylesheet" type="text/css" href="stylesheets/styles.css">
+</head>
+<body>
+<div class="container">
+<?php
```
Comment (sjr-mayden) [inline addNewGuitar.php]: Can all be on the same line

File: functions.php [hunk 9]
```diff
@@ -78,3 +78,124 @@ function displayGuitars(array $guitarsArray):string {
     }
     return $result;
     }
+
+function checkBrands(\PDO $database):bool {
```
Comment (sjr-mayden) [inline functions.php:82]: For the future - DEFINITELY NOT NOW - a transaction might be useful here as all the queries seem to be needed together

Comment (sjr-mayden) [reply functions.php:82]: Or take the fieldname and table as a parameter into a checkForAThing function

File: addNewGuitar.php [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+require_once 'functions.php';
+
+$db = getDB();
+
+$message = '<p>I want to add a new guitar for you!</p>';
+$brandInDB = checkBrands($db);
```
Comment (sjr-mayden) [inline addNewGuitar.php]: These function calls are causing errors, need to be protected when the page is opened for the first time

File: addNewGuitar.php [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+    <link rel="stylesheet" type="text/css" href="stylesheets/normalize-8-0-1.css">
+    <link rel="stylesheet" type="text/css" href="stylesheets/styles.css">
+</head>
+<body>
+<div class="container">
+<?php
+    echo $message;
+    ?>
+    <form action="addNewGuitar.php" method="get" class="form-container">
+        <div class="form-item">Brand: <input type="text" name="brand" required></div>
```
Comment (sjr-mayden) [inline addNewGuitar.php:54]: Is the fact these need to be not NULL validated in the PHP?

**Review by sjr-mayden — CHANGES_REQUESTED** [review]

File: addNewGuitar.php (hunk 11 above)
Comment (sjr-mayden) [reply addNewGuitar.php:54]: Yes

**Review by sjr-mayden — APPROVED** [review]

**Pull request merged**